src/test/com/fulcrologic/fulcro/algorithms/normalized_state_helpers_spec.cljc

```diff
@@ -3,33 +3,205 @@
+
+;; TODO Make sure that a path that does not go through idents still resolves correctly (in case of denormalized state)
+
+
+
+(specification "tree-path->db-path"
+  (behavior "In normalized DB, when root key is a to-many list of idents"
+    (let [state normalized-state]
       (assertions
-        (nsh/get-in state [:a :c :value]) => 42)))
-  (behavior "Follows edges from root that are denormalized"
-    (let [state {:a {:c {:value 42}}}]
+        (nsh/tree-path->db-path state [:person/id 1 :person/spouse :person/name])
+        => [:person/id 2 :person/name])))
+
+  (behavior "In normalized DB, follows idents ending with to-many relations"
+    (let [state normalized-state]
+      (assertions
+        (nsh/tree-path->db-path state [:person/id 1 :person/spouse :person/cars :car/id])
+        => [:person/id 2 :person/cars :car/id])))
+
+  ;; FIXME
+  (behavior "In normalized DB, follows idents ending with to-many relations"
+    (let [state normalized-state]
       (assertions
-        (nsh/get-in state [:a :c :value]) => 42)))
-  ;; What other cases can you think of?  What about the support for not-found?, etc.
+        (nsh/tree-path->db-path state [:person/id 1 :person/spouse :person/cars :car/id])
+        => [:person/id 2 :person/cars :car/id])))
+
+  #_(behavior "In denormalized DB, follows idents ending with to-many relations"
+              (let [state denormalized-state]
+                ;; FIXME [:table-key 3 :to-one-field :attr]
+                (assertions
+                  (nsh/tree-path->db-path state [:car/id 1 :car/model])
+                  => [:person/id 2 :person/first-name])))
+
+  #_(behavior "In denormalized DB, should return the provided tree-path"
+              (let [state denominator]
+                (assertions
+                  (nsh/tree-path->db-path state [:root :person/id :person/spouse :person/cars :car/id])
+                  => [:root :person/id :person/spouse :person/cars :car/id])))
+
+  #_(behavior "In denormalized DB, should return not-found when fails to resolve the given tree-path"
+              (let [state normalized-state]
+                (assertions
+                  (nsh/tree-path->db-path state [:root :person/id :person/spouse :car/id])
+                  => [:root :person/id :person/spouse :person/cars :car/id])))
+
   )
 
-;; tests others
 
 (comment
-  (def env {:state (atom {})})
-  (nsh/swap!-> env
-    (assoc :x 1)
-    (update :x inc))
 
-  env
+
+  '())
+
+
+;;============================================================================
+
+(specification "get-in"
+  (behavior "Follows edges from root that are denormalized"
+    (let [state normalized-state]
+      (assertions
+        (nsh/get-in state [:car/id 1]) => "T-150")))
   )
+
+
+(comment
+
+  '())
+
+
+;;============================================================================
+
+;(specification "remove-entity*"
+;  (behavior "1"
+;    (let [state {:a [:b 1]
```

Comment:
This was a valid get-in test..why did you drop it?
The test you have on line 111 is not right..you'd never find a string in a table entry.

Reply:
Oh, no it's not gone. Just that I've commented things to focus on certain tests only.

Reply:
To focus tests, add `:focus` after the name:
```
(specification "my test" :focus ...)
```

Then use this to run them (add a new REPL command and give it diff shortcut):

```
(require 'fulcro-spec.reporters.repl)
(in-ns (.getName *ns*))
(when (System/getProperty "test")
 (fulcro-spec.reporters.repl/run-tests #(:focus (meta %))))
```

Reply:
Ah, this is a useful one - thanks!